include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -3078,7 +3079,7 @@ namespace internal
 
         // Distinguish the inner loop based on whether we have a
         // renumbering or not
-        if (renumber.empty())
+        if (no_renumber || renumber.empty())
```

Comment:
I'd be curious here as well as to why you chose to add this. This is not the innermost loop (where we do at least 3 FMAs), so the only chance I see this making a difference is due to better code generation when the function is shorter. Is that the case? Otherwise, a single branch out of the reduction in x direction looks very cheap, which is why I chose it back then. On the other hand, many things have changed over the last 2 years, so things that are uncritical need to be reassessed.

Reply:
I did not analyze the machine code in detail but I suspect that the shorter function helps the compiler to generate better code. I think this will especially be relevant for the values only functions from https://github.com/dealii/dealii/pull/15273 where we do even less work in the loops.

Reply:
Sounds ok, let's go with the proposed implemenation.